src/SmartComponents/BaselinesPage/DeleteBaselinesModal/DeleteBaselinesModal.js

```diff
@@ -0,0 +1,98 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { Modal, Button } from '@patternfly/react-core';
+
+import { baselinesTableActions } from '../../BaselinesTable/redux/index';
+
+class DeleteBaselinesModal extends Component {
+    constructor(props) {
+        super(props);
+        this.deleteBaselines = this.deleteBaselines.bind(this);
+
+        this.state = {
+            modalOpened: this.props.modalOpened
+        };
+
+        this.toggleModal = () => {
+            const { modalOpened } = this.state;
+
+            this.setState({ modalOpened: !modalOpened });
+        };
+    }
+
+    async deleteBaselines() {
+        const { clearSelectedBaselines, deleteSelectedBaselines, selectedBaselineIds, fetchBaselines, baselineId } = this.props;
+        let apiBody;
+
+        /*eslint-disable camelcase*/
+        if (baselineId) {
+            apiBody = { data: { baseline_ids: [ baselineId ]}};
+        } else {
+            apiBody = { data: { baseline_ids: selectedBaselineIds }};
+        }
+        /*eslint-enable camelcase*/
+
+        this.toggleModal();
+        clearSelectedBaselines();
```

Comment:
If we clear and then the delete fails, I feel like we'd be in a bad state.

Reply:
I think the issue with not deselecting is that it would still be full of the uuids of the deleted baselines. I think we have to clear it, but maybe we check the success status of the delete first.

Reply:
The only "bad state" would be that the person would have to reselect all of the baselines.